frontend/lib/resolution/resolution-error-classes-list.cpp

```diff
@@ -328,6 +328,14 @@ void ErrorDotExprInUseImport::write(ErrorWriterBase& wr) const {
       "'use' or 'import'.");
 }
 
+void ErrorDotTypeOnType::write(ErrorWriterBase& wr) const {
+  auto dot = std::get<const uast::Dot*>(info);
+  auto dottedType = dot->receiver();
+  wr.heading(kind_, type_, dot, "can't apply '.type' to a type (",
```

Comment:
I think you ought to be careful here, since `dottedType` could be a long function call, and not just `t`. I suggest putting the expression printing into a `wr.message` and excluding it from the brief message output. Alternatively, you already print it in the `wr.code` call below.

Reply:
Makes sense. What if we instead pass in the resolved type of `dottedType` to this error so we could print out just a type name?

Reply:
That would be good too, but perhaps in addition to what's being printed already. For example:

```
─── error in dotTypeOnType.chpl:4 [DotTypeOnType] ───
  Can't apply '.type' to a type.
      |
    4 | type tType = t.type;
      |              ⎺⎺⎺⎺⎺⎺
      |
  The '.type' accessor can only be applied to values, but the receiver of the above expression is the type 'int'.
```

If you pass a (possibly empty) `toId`, you could even print more information if `t` is a variable (like it is in your case).
```
─── error in dotTypeOnType.chpl:4 [DotTypeOnType] ───
  Can't apply '.type' to a type.
      |
    4 | type tType = t.type;
      |              ⎺⎺⎺⎺⎺⎺
      |
  The '.type' accessor can only be applied to values, but the receiver of the above expression is the type 'int'.
  The receiver is declared as a 'type' variable here:
      |
    1 | type t = int;
      |              ⎺⎺⎺⎺⎺⎺
      |
```